From 75aaade16b00ff519fbaedb4cc773b654c11a34a Mon Sep 17 00:00:00 2001
From: tblume <Thomas.Blume@suse.com>
Date: Thu, 1 Feb 2018 22:46:15 +0100
Subject: [PATCH] meson.build: make docdir configurable (#8068)

SUSE is using a different docdir, so the path should be configurable
---
 meson.build       | 6 +++++-
 meson_options.txt | 2 ++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 36a62d280d..e497924cb8 100644
--- a/meson.build
+++ b/meson.build
@@ -136,13 +136,17 @@ udevhwdbdir = join_paths(udevlibexecdir, 'hwdb.d')
 catalogdir = join_paths(prefixdir, 'lib/systemd/catalog')
 kernelinstalldir = join_paths(prefixdir, 'lib/kernel/install.d')
 factorydir = join_paths(datadir, 'factory')
-docdir = join_paths(datadir, 'doc/systemd')
 bootlibdir = join_paths(prefixdir, 'lib/systemd/boot/efi')
 testsdir = join_paths(prefixdir, 'lib/systemd/tests')
 systemdstatedir = join_paths(localstatedir, 'lib/systemd')
 catalogstatedir = join_paths(systemdstatedir, 'catalog')
 randomseeddir = join_paths(localstatedir, 'lib/systemd')
 
+docdir = get_option('docdir')
+if docdir == ''
+        docdir = join_paths(datadir, 'doc/systemd')
+endif
+
 dbuspolicydir = get_option('dbuspolicydir')
 if dbuspolicydir == ''
         dbuspolicydir = join_paths(datadir, 'dbus-1/system.d')
diff --git a/meson_options.txt b/meson_options.txt
index 39822d6cdc..48ee3f52a4 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -134,6 +134,8 @@ option('pamlibdir', type : 'string',
        description : 'directory for PAM modules')
 option('pamconfdir', type : 'string',
        description : 'directory for PAM configuration ["no" disables]')
+option('docdir', type : 'string',
+       description : 'documentation directory')
 
 option('fallback-hostname', type : 'string', value : 'localhost',
        description : 'the hostname used if none configured')
